### ISSUE TITLE
App crashes on Emulators with SDK 31 and SDK 32

### ISSUE THREAD

**Issue by rafaelekol** [zcash/secant-android-wallet#430]
When I try to run this app on emulator with SDK 31 and higher it throws error that you can see below. 

```
 java.lang.UnsatisfiedLinkError: No implementation found for java.lang.String[][] cash.z.ecc.android.sdk.tool.DerivationTool.deriveUnifiedViewingKeysFromSeed(byte[], int, int) (tried Java_cash_z_ecc_android_sdk_tool_DerivationTool_deriveUnifiedViewingKeysFromSeed and Java_cash_z_ecc_android_sdk_tool_DerivationTool_deriveUnifiedViewingKeysFromSeed___3BII)
        at cash.z.ecc.android.sdk.tool.DerivationTool.deriveUnifiedViewingKeysFromSeed(Native Method)
        at cash.z.ecc.android.sdk.tool.DerivationTool.access$deriveUnifiedViewingKeysFromSeed(DerivationTool.kt:8)
        at cash.z.ecc.android.sdk.tool.DerivationTool$Companion.deriveUnifiedViewingKeysFromSeed(Unknown Source:0)
        at cash.z.ecc.android.sdk.tool.DerivationTool$Companion.access$deriveUnifiedViewingKeysFromSeed(DerivationTool.kt:10)
        at cash.z.ecc.android.sdk.tool.DerivationTool$Companion$deriveUnifiedViewingKeys$2.invoke(DerivationTool.kt:23)
        at cash.z.ecc.android.sdk.tool.DerivationTool$Companion$deriveUnifiedViewingKeys$2.invoke(DerivationTool.kt:22)
        at cash.z.ecc.android.sdk.tool.DerivationTool$Companion.withRustBackendLoaded(DerivationTool.kt:109)
        at cash.z.ecc.android.sdk.tool.DerivationTool$Companion.access$withRustBackendLoaded(DerivationTool.kt:10)
        at cash.z.ecc.android.sdk.tool.DerivationTool$Companion$withRustBackendLoaded$1.invokeSuspend(Unknown Source:15)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
        at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:104)
        at android.os.Handler.handleCallback(Handler.java:938)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loopOnce(Looper.java:201)
        at android.os.Looper.loop(Looper.java:288)
        at android.app.ActivityThread.main(ActivityThread.java:7839)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1003)
```

**Comment by HonzaR** [comment]
@rafaelekol thank you again for reporting the issue. @ccjernigan so after some research and testing I've found the problematic past change, which causes the unexpected behavior.

- It's happening inside the SDK project - so I'll file the new issue there.
- It's related to `x86_64` architecture, we can be sure, because emulator with SDK API 30 with `x86` works well, but the `x86_64` suffers with the crash.
- The crash happens, when we try to get addresses with use of the method `deriveUnifiedViewingKeysFromSeed`, which is not known for `x86_64` rust code inside the SDK project.
- The Payment app suffers with the crash after persisting a mnemonic and initializing the wallet object.
- The demo-app from SDK also suffers with the crash when we try to open Get Address from side menu.  
- It started with the changes of this [commit](https://github.com/zcash/zcash-android-wallet-sdk/commit/21f67ef23acbb80409d52371226715e9167f301d). There is a change in `ndkVersion`, which causes the trouble. Reverting it back, works well.

**Comment by ccjernigan** [comment]
@HonzaR Thanks for investigating!  Can you confirm that the problem goes away if we revert only the NDK version but keep the Mozilla Gradle Plugin version at 0.9.3?

We changed both at the same time, since newer NDK support was a feature of Mozilla's Rust Gradle Plugin 0.9.3.  If reverting just the NDK works, then that'll be our short-term solution.

**Comment by HonzaR** [comment]
> @HonzaR Thanks for investigating! Can you confirm that the problem goes away if we revert only the NDK version but keep the Mozilla Gradle Plugin version at 0.9.3?

Yes, I can confirm that the reverting only the NDK version back to to previous version, as the commit shows, is enough to have the SDK project work for me again. So keep the  Mozilla Gradle Plugin and revert the NDK version is the way. Do you want me to make this change?

**Comment by ccjernigan** [comment]
Yes, please.  That would be great.

**Comment by HonzaR** [comment]
The problem with the SDK is partially solved by reverting its NDK version back to `22.1.7171670`. Once the SDK version `1.6.0-beta01` will be released, the problem with `x86_64` emulators will be solved.
